addon/background.js

```diff
@@ -105,22 +99,35 @@ chrome.webRequest.onHeadersReceived.addListener(function (info) {
   return {};
 }, requestFilter, ["blocking", "responseHeaders"]);
 
-function retry(attempter, options) {
+async function retry(attempter, options) {
   let times = options.times || 3;
   let wait = options.wait || 100;
-  return attempter().catch((error) => {
+  try {
+    return await attempter();
+  } catch (error) {
     times--;
     if (times <= 0) {
       throw error;
     }
-    return timeout(wait).then(() => {
-      retry(attempter, {times, wait});
-    });
-  });
+    await timeout(wait);
+    return retry(attempter, {times, wait});
+  }
 }
 
 function timeout(time) {
   return new Promise((resolve) => {
     setTimeout(resolve, time);
   });
 }
+
+async function init() {
+  try {
+    const result = browser.storage.sync.get("desktopHostnames");
```

Comment:
Doesn't this have to be `const result = await browser.storage.sync.get("desktopHostnames");` ?